File: src/01-easy/the-river-i-/solution.swift
func ds(_ n:Int)->Int{var x=n,s=0;while x>0{s+=x%10;x/=10};return s}
var r1=Int(readLine()!)!,r2=Int(readLine()!)!
while r1 != r2{if r1<r2{r1+=ds(r1)}else{r2+=ds(r2)}}
print(r1)

File: src/01-easy/the-river-i-/solution.ts
const ds=(n:number):number=>[...String(n)].reduce((s,c)=>s+ +c,0);
let r1=+readline(),r2=+readline();
while(r1!==r2){if(r1<r2)r1+=ds(r1);else r2+=ds(r2);}
console.log(r1);
